phoenix-core/src/main/java/org/apache/phoenix/compile/UpsertCompiler.java

```diff
@@ -292,7 +292,8 @@ public MutationPlan compile(UpsertStatement upsert) throws SQLException {
                 // Cannot update:
                 // - read-only VIEW 
                 // - transactional table with a connection having an SCN 
-                if ( table.getType() == PTableType.VIEW && table.getViewType().isReadOnly() ) {
+                if ( ( table.getType() == PTableType.VIEW && table.getViewType().isReadOnly() ) 
+                     || ( table.isTransactional() && connection.getSCN() != null )) {
```

Comment:
Same thing here - separate, specific exception for transactional tables with connection SCN.